packages/sanity/src/form/inputs/arrays/ArrayOfObjectsInput/ArrayInput.tsx

```diff
@@ -284,12 +284,13 @@ export class ArrayInput extends React.PureComponent<ArrayInputProps> {
   }
 
   renderMember(member: ArrayOfObjectsMember) {
-    const {renderField, renderInput, renderPreview} = this.props
+    const {renderField, renderInput, renderPreview, renderItem} = this.props
     if (member.kind === 'item') {
       return (
         <ArrayOfObjectsItem
           member={member}
-          renderItem={this.renderItem}
+          // eslint-disable-next-line react/jsx-no-bind
+          renderItem={(p) => this.renderArrayItem({...p, children: renderItem(p)})}
```

Comment:
Agree, this can be handled better. But I will wait fixing this since we are working on refactoring the array inputs to functional components where this can be improved!